unidist/core/backends/mpi/core/shared_object_store.py

```diff
@@ -190,9 +190,8 @@ def _allocate_shared_memory(self):
         self.service_shared_buffer = memoryview(service_buffer).cast("l")
         # Set -1 to the service buffer because 0 is a valid value and may be recognized by mistake.
         if mpi_state.is_monitor_process():
-            self.service_shared_buffer[:] = array(
-                "l", [-1] * len(self.service_shared_buffer)
-            )
+            memoryview_array = np.frombuffer(self.service_shared_buffer, dtype=np.int64)
+            memoryview_array[:] = -1
```

Comment:
```suggestion
            values_to_fill = array("l", [-1] * len(self.service_shared_buffer))
            values_to_fill_view = memoryview(values_to_fill)
            parallel_memcopy(
                values_to_fill_view,
                self.self.service_shared_buffer[:len(self.service_shared_buffer)],
                6,
            )
```
To not introduce a new dependency (numpy) let's try to use our own function to set the value.

Reply:
This is as slow as the previous approach. Let's discuss offline what we should do.